### DISCUSSION TITLE
HTMX custom headers are not being passed through NGINX

### DISCUSSION THREAD

**Discussion by lonnieezell** [laravel/valet#1493] [General]
I have a site I'm working on that's using HTMX. HTMX can use a number of custom headers in the request and I'm trying to capture those values to output them in a debug screen. However, they don't appear to be making it through the server to the PHP script. 

Looking at the request in the browser it clearly lists the following headers being passed: 

- Hx-Boosted
- Hx-Current-Url
- Hx-Request

Once in PHP-land there none of those are showing up when I look at `$_SERVER` or using `getallheaders()`. 

I remember having a similar issue years ago with Apache and was fixed with a htaccess rule, IIRC, but I haven't run across this using Nginx. I've spent a couple hours searching and I guess my Google-fu is failing today. 

Anyone know of a configuration value for Nginx that might be causing these custom headers to be lost? 

Thanks.

**Comment by drbyte** [comment]
I wonder if you just need to explicitly pass those parameters from nginx to PHP via `fastcgi_params` configuration?

Upon install, Valet updates the default nginx `fastcgi_params` from this stub:
https://github.com/laravel/valet/blob/master/cli/stubs/fastcgi_params
in `/usr/local/etc/nginx/fastcgi_params` (or /opt/homebrew/...)

Or if it really needed to be specific to a certain site, you could edit that site's `.conf` file in `~/.config.valet/nginx/*.conf`
